src/components/VideoCard/VideoCard.jsx

```diff
@@ -77,7 +98,33 @@ function VideoCard({
               <span className="text-muted">{getLastPosted(releaseDate)}</span>
             </div>
           </div>
-          <div className="video-menu">{getIcons('VIDEO_MENU', '20px')}</div>
+          <div className="video-menu">
+            <span
+              onClick={e => {
+                e.stopPropagation();
+                setDisplay(prevState => !prevState);
+              }}
+            >
+              {getIcons('VIDEO_MENU', '20px')}
+            </span>
+            <div
+              className={display ? 'menu-options show-menu' : 'menu-options '}
```

Comment:
as `menu-options` is the class which is common you could write this as
```jsx
className={`menu-options ${display ? "show-menu" : ""}`}
```

This follows the DRY(do not repeat yourself) method